src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -754,6 +758,49 @@ public override bool Equals(ConstantValue? other)
             }
         }
 
+        private sealed class ConstantValueNativeInt : ConstantValueDiscriminated
+        {
+            private readonly int _value;
```

Comment:
>int [](start = 29, length = 3)

nit: may be worth a comment (32bit is the common denominator between nints on different platforms)